test/contracts/option-pair-factory.test.ts

```diff
@@ -0,0 +1,140 @@
+import chai from 'chai';
+import {ethers} from '@nomiclabs/buidler';
+import {Signer} from 'ethers';
+import {deploy0, getCreatePairEvent} from '../../lib/contracts';
+import {getTimeNow} from '../common';
+import {ErrorCode} from '../../lib/constants';
+import {OptionPairFactoryFactory, OptionFactory} from '../../typechain';
+import {OptionPairFactory} from '../../typechain/OptionPairFactory';
+import {deployMockContract, MockContract} from 'ethereum-waffle';
+import ERC20Artifact from '../../artifacts/ERC20.json';
+import BTCRefereeArtifact from '../../artifacts/BTCReferee.json';
+import {
+  getCreate2OptionAddress,
+  getCreate2ObligationAddress
+} from '../../lib/addresses';
+import {newBigNum} from '../../lib/conversion';
+import {option} from 'fast-check';
+
+const {expect} = chai;
+
+describe('OptionPairFactory.sol', () => {
+  let alice: Signer;
+  let bob: Signer;
+
+  let aliceAddress: string;
+  let bobAddress: string;
+
+  let optionFactory: OptionPairFactory;
+  let collateral: MockContract;
```

Comment:
Can we add a second ERC20 test contract that uses `uint256` for the decimal type? USDT is using this type (see https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7#readContract). Just so we make sure that our handling of decimals also works with their contract.

Reply:
On second thought, we will get this for free with #41, what do you think?